### ISSUE TITLE
Integrating AJAX and Dropdown module

### ISSUE THREAD

**Issue by aprog** [Semantic-Org/Semantic-UI#1789]
Hi,

I'm using Dropdown module and it works well until I apply it to select element that has AJAX behaviour.

Here is a short description in Drupal context: after the page was loaded I initialize Dropdown on each select element. After the user selected some value Drupal sends a request to the backed, backend answers with updated form. The response includes select element with some pre-selected value.

After that we initialize Dropdown on the select element and Semantic UI sets value to the Dropdown and triggers ".change" JavaScript event. This event causes AJAX request to the backend. This produces infinite AJAX calls.

Also I noticed that Semantic UI will set value and trigger ".change" event if select element has pre-selected value.

It happened after updating framework to the latest version (1.8.1). I guess that initialize method was updated to use module.set.selected() which calls module.set.value() and triggers ".change" event.

I'm just curious are there any ways to not trigger ".change" event when the select element already has this value? (I guess it will not cause AJAX call to the backend). Theoretical solution may be updating module.set.value() to something like this:

```javascript
value: function(value) {
  module.debug('Adding selected value to hidden input', value, $input);
  if($input.length > 0 && value != $input.val()) {
    $input
      .val(value)
      .trigger('change')
    ;
  }
  else {
    $module.data(metadata.value, value);
  }
},
```

But I'm not sure will it work in case if user selected some value and refreshed the page.

Thanks in advance

**Comment by jlukic** [comment]
I appreciate the thorough description but a test case would really help me understand.

**Comment by aprog** [comment]
Hi Jack,

thanks for paying attention to this issue.

I prepared a little test case that shows the problem: http://jsfiddle.net/jaev39pL/9/.

Let me describe the algorithm in few words:
1. after the page was loaded I initialize Dropdown module
2. after the user selected some value I send the form to the backend and get updated form with pre-selected value. By default select element is hidden, so we need to initialize Dropdown module on it
3. during initialization Semantic UI triggers change event that causes AJAX request to the backend
4. things repeat from step 2 (you may notice infinite AJAX requests in the Firebug/Development tools console)

Please let me know if you need more information.

**Comment by jlukic** [comment]
Here's an example of how you can update options
http://jsfiddle.net/2avsd0en/

**Comment by jlukic** [comment]
If you for some reason really had to replace the `<select>` itself and not just options. You will have to refresh the menu. This is because the observers are attached to the select element, and removing from the DOM will not allow the changes to be observed.

Here's an example of that:
http://jsfiddle.net/2v3ktaLr/

**Comment by aprog** [comment]
Thanks for examples.

They are helpful for case when a developer is creating a form from scratch. In my case I have already written system that provides AJAX integration for forms. For example, system automatically returns and updates entire element. Or in some cases backend may return entire form. But anyway thank you for the suggestions.

I noticed that this task was added to the 1.9.0 milestone, does it mean that you are going to continue work on this task?
